hw/ip/aes/doc/aes.md

```diff
@@ -0,0 +1,296 @@
+      {  name: 'addr',        wave: 'x2345xxxx|xxxxxxx', data: 'K0 K1 K2 K3'},
+      {  name: 'wdata',       wave: 'x2345xxxx|xxxxxxx', data: 'K0 K1 K2 K3'},
+    ],
+    {},
+    ['AES Unit',
+      {  name: 'Config mode', wave: 'x4...............', data: 'DECRYPT'},
+      {  name: 'AES mode',    wave: '2........|.4.....', data: 'IDLE DECRYPT'},
+      {  name: 'KEM mode',    wave: '2....3...|.4.....', data: 'IDLE ENCRYPT DECRYPT'},
+      {  name: 'round',       wave: 'xxxxx2.22|22.2222', data: '0 1 2 9 0 1 2 3 4'},
+      {  name: 'key_init',    wave: 'xxxx5....|.......', data: 'K0-3'},
+      {  name: 'key_full',    wave: 'xxxxx5222|4.22222', data: 'K0-3 f(K) f(K) f(K) K0-3\' f(K) f(K) f(K) f(K) f(K)'},
+      {  name: 'key_dec',     wave: 'xxxxxxxxx|4......', data: 'K0-3\''},
+    ]
+  ]
+}
+```
+
+The AES unit is configured to do decryption (`Config mode` = DECRYPT).
+Once the new key has been provided via the control and status registers (top), this new key is loaded into the Full Key register (`key_full` = K0-3) and the KEM starts performing encryption (`KEM mode`=ENCRYPT).
+The cipher data path remains idle (`AES mode`=IDLE).
+In every round, the value in `key_full` is updated.
+After 10 encryption rounds, the value in `key_full` equals the start key for decryption.
+This value is stored into the Decryption Key register (`key_dec` = K0-3’ at the very bottom).
+Now the AES unit can switch between encryption/decryption without overhead as both the start key for encryption (`key_init`) and decryption (`key_dec`) can be loaded into `full_key`.
+
+For details on the KeyExpand operation refer to the [AES specification, Section 5.2](https://csrc.nist.gov/csrc/media/publications/fips/197/final/documents/fips-197.pdf).
+
+Key expanding is the only operation in the AES unit for which the functionality depends on the selected key length.
+Having a KEM that supports 128-bit key expansion, support for the 256-bit mode can be added at low overhead.
+In contrast, the 192-bit mode requires much larger muxes.
+Support for this mode is thus optional and can be enabled/disabled via a design-time parameter.
+
+Once we have cost estimates in terms of gate count increase for 192-bit mode, we can decide on whether or not to use it in OpenTitan.
+Typically, systems requiring security above AES-128 go directly for AES-256.
+
+
+{{% section2 Hardware Interfaces }}
+
+{{% hwcfg Name }}
+
+
+{{% section2 Design Details }}
+
+This section discusses different design details of the AES module.
+
+
+### System Key-Manager Interface
+
+This version of the AES unit is controlled entirely by the processor.
+The processor writes both input data as well as the initial key to dedicated registers via the system bus interconnect.
+
+Future versions of the AES unit might include a separate interface through which a possible system key manager can update the initial key without exposing the key to the processor or other hosts attached to the system bus interconnect.
```

Comment:
same recommendation as before:
^can update the initial key without exposing the key^can provide the key without exposing it^

Reply:
Done.